meilisearch-lib/src/tasks/task_store/store.rs

```diff
@@ -161,18 +162,25 @@ impl Store {
                 index_uid.push(0);
                 self.uids_task_ids
                     .remap_key_type::<ByteSlice>()
-                    .prefix_iter(txn, &index_uid)?
-                    .try_fold(
-                        &mut candidates,
-                        |candidates, entry| -> StdResult<_, heed::Error> {
-                            let (key, _) = entry?;
-                            let (_, id) = IndexUidTaskIdCodec::bytes_decode(key)
-                                .ok_or(heed::Error::Decoding)?;
-                            candidates.insert(id);
-
-                            Ok(candidates)
-                        },
-                    )?;
+                    .rev_prefix_iter(txn, &index_uid)?
+                    .map(|entry| -> StdResult<_, heed::Error> {
+                        let (key, _) = entry?;
+                        let (_, id) =
+                            IndexUidTaskIdCodec::bytes_decode(key).ok_or(heed::Error::Decoding)?;
+                        Ok(id)
+                    })
+                    .take_while(|entry| {
+                        entry
+                            .as_ref()
+                            .ok()
+                            // as soon as we are out of the range we exit
+                            .map(|key| !range.contains(key))
```

Comment:
uhmm, what if the first element is out of the range? I think you mean to filter here?